Update former uses of jQuery.trim to work properly with non-string types

jonathansadowski opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR updates the changes that were made in #29044 for jQuery compatibility. The issue is that `jQuery.trim()` worked on types other than `string`. In a lot of the replacements we made in that PR, we were checking explicitly for type of `string` and returning an empty string when the type wasn't string.

In cases where we were previously using `jQuery.trim( a )`, this PR replaces those instances with `( a || 0 == a ) ? a.toString().trim() : ''`. This handles the following cases in the same manner as `jQuery.trim`:
* `null`
* `undefined`
* `false`
* Numbers, including 0, Infinity, etc.
* Objects
* Arrays
* Anything else that has a `.toString()` method.

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #30069.

### How to test the changes in this Pull Request:

1. Install and configure the Authorize.net Payment Gateway
2. Verify that you're able to checkout using the Gateway
3. Test the changes in select2/selectwoo using the testing plugin from this PR: https://github.com/woocommerce/woocommerce/pull/29989

Hat-tip to @juagonala for the PR to fix the issue in #29989. I wanted to keep the fix consistent with the fixes for the payment gateway, so I've included a different changeset for the issue in this PR.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Fix - Update jQuery compatibility changes to trim functions to work with types other than string.

Commit:
Update former uses of jQuery.trim to work properly with non-string types

Commit:
Fix a typo in params.term comparison

Commit:
Restore inadvertant removal of semicolon in select2

roykho commented on assets/js/select2/select2.full.js:
```diff
@@ -4941,7 +4941,7 @@ S2.define('select2/defaults',[
     function matcher (params, data) {
       // Always return the object if there is nothing to compare
-      if ( params.term.trim() === '' ) {
+      if ( params.term == null || params.term.toString().trim() === '' ) { 
```
Coding standards: Need to remove the space after `{`

roykho commented on assets/js/selectWoo/selectWoo.full.js:
```diff
@@ -6131,7 +6131,7 @@ S2.define('select2/compat/matcher',[
       var match = $.extend(true, {}, data);
 
-      if ( params.term == null || ( 'string' === typeof params.term && '' === params.term.trim() ) ) {
+      if (params.term == null || params.term.trim() === '') {
```
Coding standards: Add space after opening `(` and before closing `)`

roykho reviewed (dismissed):
Tested Authorize.net and various Select2 single/multi select "dropdowns" and they're working.

Left small nitpick comments for coding standards. Otherwise LGTM. Feel free to merge after fixes.

Commit:
Update for coding standards

jonathansadowski commented:
> Left small nitpick comments for coding standards. Otherwise LGTM. Feel free to merge after fixes.

Thanks — addressed the issues with the coding standards. I'll merge once the tests pass.
